Example/src/CubesExample.tsx

```diff
@@ -0,0 +1,360 @@
+    0,
+    2 * x * z - 2 * s * y,
+    2 * y * z + 2 * s * x,
+    1 - 2 * x * x - 2 * y * y,
+    0,
+    0,
+    0,
+    0,
+    1,
+  ];
+}
+
+function eulerToQuaternion(pitch: number, roll: number, yaw: number) {
+  'worklet';
+  const c1 = Math.cos(pitch * 0.5);
+  const s1 = Math.sin(pitch * 0.5);
+  const c2 = Math.cos(roll * 0.5);
+  const s2 = Math.sin(roll * 0.5);
+  const c3 = Math.cos(yaw * 0.5);
+  const s3 = Math.sin(yaw * 0.5);
+
+  return [
+    s1 * c2 * c3 - c1 * s2 * s3,
+    c1 * s2 * c3 + s1 * c2 * s3,
+    c1 * c2 * s3 + s1 * s2 * c3,
+    c1 * c2 * c3 - s1 * s2 * s3,
+  ];
+}
+
+function multiplyQuaternions(a: number[], b: number[]) {
+  'worklet';
+  return [
+    a[3] * b[0] + a[0] * b[3] + a[1] * b[2] - a[2] * b[1],
+    a[3] * b[1] - a[0] * b[2] + a[1] * b[3] + a[2] * b[0],
+    a[3] * b[2] + a[0] * b[1] - a[1] * b[0] + a[2] * b[3],
+    a[3] * b[3] - a[0] * b[0] - a[1] * b[1] - a[2] * b[2],
+  ];
+}
+
+const sidesRotations = [
+  [0, 0, 0],
+  [Math.PI / 2, 0, 0],
+  [-Math.PI / 2, 0, 0],
+  [0, Math.PI / 2, 0],
+  [0, -Math.PI / 2, 0],
+  [Math.PI, 0, 0],
+];
+
+const sidesColors = [
+  '#8B9576',
+  '#0D519A',
+  '#AF351B',
+  '#555E6B',
+  '#7F09E1',
+  '#9DB89A',
+];
+
+function CubeWithEulerAngles() {
+  const animatedSensor = useAnimatedSensor(SensorType.ROTATION, {
+    iosReferenceFrame: IOSReferenceFrame.XArbitraryZVertical,
+    adjustToInterfaceOrientation: true,
+  });
+
+  const sidesStyles = sidesRotations.map((rotation, i) =>
+    useAnimatedStyle(() => {
+      const pitch = animatedSensor.sensor.value.pitch;
+      const roll = animatedSensor.sensor.value.roll;
+      const yaw = animatedSensor.sensor.value.yaw;
+
+      const sideLength = 100;
+      const origin = { x: 0, y: 0, z: -sideLength / 2 };
+
+      const it = eulerToMatrix(rotation[0], rotation[1], rotation[2]);
+      const matrix = eulerToMatrix(pitch, -roll, yaw);
+      multiplyInto(matrix, matrix, it);
+      transformOrigin(matrix, origin);
+
+      return {
+        transform: [{ perspective: 1000 }, { matrix: matrix }],
+        backgroundColor: sidesColors[i],
+      };
+    })
+  );
+
+  return (
+    <View style={componentStyle.boxContainer}>
+      {sidesStyles.map((style, i) => (
+        <Animated.View style={[componentStyle.box, style]} key={i}>
+          <Text>EULER</Text>
+        </Animated.View>
+      ))}
+    </View>
+  );
+}
+
+function CubeWithQuaternions() {
+  const animatedSensor = useAnimatedSensor(SensorType.ROTATION, {
+    adjustToInterfaceOrientation: true,
+  });
+
+  const sidesStyles = sidesRotations.map((rotation, i) =>
+    useAnimatedStyle(() => {
+      const sideLength = 100;
+      const origin = { x: 0, y: 0, z: -sideLength / 2 };
+
+      const q0 = eulerToQuaternion(-rotation[0], -rotation[1], rotation[2]);
+
+      const q1 = [
+        -animatedSensor.sensor.value.qx,
+        animatedSensor.sensor.value.qy,
+        -animatedSensor.sensor.value.qz,
+        animatedSensor.sensor.value.qw,
```

Comment:
Let's assign these four values beforehand, i.e.
```js
const [qx, qy, qz, qw] = animatedSensor.sensor.value;
const q1 = [-qx, qy, -qz, qw];
```
in order to avoid calling `.value` more than once if not necessary.